File: statsig-ffi/bindings/php/src/StatsigFFI.php
<?php

namespace Statsig\StatsigFFI;

use FFI;

class StatsigFFI
{
    private static ?FFI $ffi = null;
    public static function get()
    {
        if (self::$ffi === null) {
            self::$ffi = FFI::cdef(
                file_get_contents(__DIR__ . '/../../../include/statsig_ffi.h'),
                __DIR__ . '/../../../../target/release/libstatsig_ffi.dylib'
            );
        }
        return self::$ffi;
    }
}

File: statsig-ffi/bindings/ruby/statsig_ffi.rb
require 'ffi'

module StatsigFFI
  extend FFI::Library
  ffi_lib File.expand_path('./libstatsig_ffi.dylib', __dir__)

  class User < FFI::Struct
    layout :name, :string,
           :email, :string
  end

  attach_function :create_user, [:string, :string], :pointer
  attach_function :destroy_user, [:pointer], :void
  attach_function :create_statsig_for_user, [:pointer, :string], :pointer
  attach_function :destroy_statsig, [:pointer], :void
  attach_function :check_gate, [:pointer, :string], :bool
  attach_function :statsig_get_client_init_response, [:pointer], :string
end


